### PR TITLE
feat: (votes): Make proposals static and remove votes state from redux

### PR DESCRIPTION
**Pull request by memoyil**

### CODE REVIEW

File: src/pages/voting/proposal/[id].tsx [hunk 1]
```diff
@@ -1,3 +1,56 @@
...
+
+  const fetchedProposal = await getProposal(id)
+  if (!fetchedProposal) {
+    return {
+      notFound: true,
+      revalidate: 1,
+    }
+  }
+
+  const votes = await getAllVotes(fetchedProposal.id, Number(fetchedProposal.snapshot))
```
Comment (chef-jojo) [inline src/pages/voting/proposal/[id].tsx:43]: This `getAllVotes` is quite slow, when no cache available, the routing will stuck until it is finished. 🤔
https://github.com/vercel/next.js/issues/13751

We might need to sacrifice this to only load on client right now. I can think of a possible solution is save the closed proposal voting to ipfs or redis. Ongoing proposal voting should all fetch on client.

Comment (memoyil) [reply src/pages/voting/proposal/[id].tsx]: Done

**Review by chef-jojo — APPROVED** [review]
👍  just some eslint warning

File: src/pages/voting/proposal/[id].tsx [hunk 2]
```diff
@@ -1,3 +1,53 @@
...
+export const getStaticProps: GetStaticProps = async ({ params }) => {
+  const { id } = params
+  if (typeof id !== 'string') {
+    return {
+      notFound: true,
+    }
+  }
+
+  const fetchedProposal = await getProposal(id)
+  if (!fetchedProposal) {
```
Comment (chef-jojo) [inline src/pages/voting/proposal/[id].tsx:36]: try catch

Comment (memoyil) [reply src/pages/voting/proposal/[id].tsx]: Done

**Review by chef-jojo — APPROVED** [review]
lgtm and thanks, bundle size reduced

**Pull request merged**